mp/src/game/client/neo/ui/neo_ui.cpp

```diff
@@ -311,6 +327,57 @@ void EndSection()
 		}
 	}
 
+	// Scroll bar area painting and mouse interaction (no keyboard as that's handled by active widgets)
+	if (bHasScroll)
+	{
+		const int iYStart = g_pCtx->iYOffset[g_pCtx->iSection];
+		const int iYEnd = iYStart + iRowsInScreen;
+		const float flYPercStart = iYStart / (float)(g_pCtx->iWidget);
+		const float flYPercEnd = iYEnd / (float)(g_pCtx->iWidget);
+		vgui::IntRect rectHandle{
+			g_pCtx->dPanel.x + g_pCtx->dPanel.wide,
+			g_pCtx->dPanel.y + (int)(g_pCtx->dPanel.tall * flYPercStart),
+			g_pCtx->dPanel.x + g_pCtx->dPanel.wide + g_pCtx->iRowTall,
+			g_pCtx->dPanel.y + (int)(g_pCtx->dPanel.tall * flYPercEnd)
+		};
+
+		bool bAlterOffset = false;
+		switch (g_pCtx->eMode)
+		{
+		case MODE_PAINT:
+			surface()->DrawSetColor(g_pCtx->bgColor);
+			surface()->DrawFilledRectArray(&rectScrollArea, 1);
+			surface()->DrawSetColor(g_pCtx->abYMouseDragOffset[g_pCtx->iSection] ? g_pCtx->selectBgColor : g_pCtx->normalBgColor);
+			surface()->DrawFilledRectArray(&rectHandle, 1);
+			break;
+		case MODE_MOUSEPRESSED:
+			g_pCtx->abYMouseDragOffset[g_pCtx->iSection] = bMouseInScrollbar;
+			if (bMouseInScrollbar)
+			{
+				// If not pressed on handle, set the drag at the middle
+				const bool bInHandle = InRect(rectHandle, g_pCtx->iMouseAbsX, g_pCtx->iMouseAbsY);
+				g_pCtx->iStartMouseDragOffset[g_pCtx->iSection] = (bInHandle) ?
+							(g_pCtx->iMouseAbsY - rectHandle.y0) :((rectHandle.y1 - rectHandle.y0) / 2.0f);
+				bAlterOffset = true;
+			}
+			break;
+		case MODE_MOUSERELEASED:
+			g_pCtx->abYMouseDragOffset[g_pCtx->iSection] = false;
+			break;
+		case MODE_MOUSEMOVED:
+			bAlterOffset = g_pCtx->abYMouseDragOffset[g_pCtx->iSection];
+			break;
+		default:
```

Comment:
Nit. I don't think having a default is a requirement in a switch statement. Just an observation I made from reviewing your crosshair PR.

Reply:
While not a requirement, it's fine practice. On GCC (and something similar for MSVC) with `-Wall` on, `-Wswitch` turns on which will complain if an enum case hasn't been handled, where `default: break;` helps rid of that warning. I know this project have switched this warning off but I still will do `default: break;` as a habit as I've done when writing C/C++ switch statements.
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wswitch